### ISSUE TITLE
Rendering of area:highway

### ISSUE THREAD

**Issue by scaidermern** [gravitystorm/openstreetmap-carto#180]
It would be nice to see a rendering of the [area:highway](https://wiki.openstreetmap.org/wiki/Key:area:highway) key. The rendering rules are similar to the regular [highway](https://wiki.openstreetmap.org/wiki/Key:highway) key, including the highway class, except that this key applies to areas only (and should not be used for routing).

**Comment by kocio-pl** [comment]
I think this proposal:

https://wiki.openstreetmap.org/wiki/Proposed_features/Street_area

is interesting to deploy, since it's more developed at the general level, however it may lack some tagging. In the (mostly duplicate) #1298 ticket I gave the link to demo rendering script for Maperitive:

https://github.com/javnik36/highway-area-style

and additional proposition:

"I have also the idea to drop colouring by street type then (on highest zoom only) and start rendering them by surface type - I think that plus the real width and geometry would be more useful in micro scale."

**Comment by kocio-pl** [comment]
I know that we still wait for database update, but there are [some real data rendering examples added lately](https://wiki.openstreetmap.org/wiki/Proposed_features/Street_area#See_also) and we could get some ideas how should we render such areas.

**Comment by Marekkleciak** [comment]
We have for today 62418 area:highway in the map.
We have enough test cases for further discussion. Check cites Szczecin and Wrocław e.g. here:
http://osmapa.pl/w/area/?lat=51.08998&lon=17.01772&zoom=17&ol=QqEFBAG

**Comment by kocio-pl** [comment]
Starting with basics, I like general colors used on osmapa.pl, related to physical features:
- dark grey pattern for roads
- red for cycling paths
- white-and-grey pattern for footways/pedestrian area 
- light grey for traffic islands. 

Any other ideas?

**Comment by Ircama** [comment]
> * red for cycling paths

One think I would invite to consider (or at least to avoid precluding) is a future possibility to accommodate the representation of signed walking routes together with the already implemented rendering for *highways*. Being able to get over this challenge will provide new exceptional value to the map for many users. This is currently a default representation in traditional topographic outdoor maps targeted to hiking and mountain biking, where signed routes are generally represented with some kind of vivid or cardinal red color
(e.g., http://www.cicloweb.net/iti1gardavr.jpg or
http://3.bp.blogspot.com/-rChlsq-Mb1M/VoZlBPNdFqI/AAAAAAAAFDk/CUzXh5xomY0/s1600/Boccaor-Val%2Bd%2527Archeset-Val%2Bdelle%2BMure.jpg).

Related tags to be processed are here http://wiki.openstreetmap.org/wiki/Hiking and here http://wiki.openstreetmap.org/wiki/Walking_Routes. Namely, *osmc:symbol*, *ref*, *symbol*, *operator*, *network* in case of *highway=track* or *highway=path*, which should generate red routes with optional road shield and related *refs* text.

So, providing space for such color within routes (including cycling paths) would be really good IMHO.

**Comment by kocio-pl** [comment]
I think that cycling paths can be also [orange](https://i.guim.co.uk/img/static/sys-images/Guardian/Pix/pictures/2014/7/11/1405084494289/00ca3d24-861f-48f2-91ff-7837753b63b0-620x372.jpeg?w=620&q=55&auto=format&usm=12&fit=max&s=6c23e45f7a490b588d02d6dd40cdd9f7). But first I'd like to know if we plan to render any routes - I guess universal style is not about searching and guiding, rather explaining the space. 

I think we have public transport and cycling layers on OSM.org exactly because of this, we need probably another layer just for hiking there. In the meantime there are some independent specialized maps like [Hike&Bike](http://www.hikebikemap.org/) or [OpenTopoMap](https://opentopomap.org).

**Comment by Ircama** [comment]
> In the meantime there are some independent specialized maps like Hike&Bike or OpenTopoMap.

Yes I know. I really admire the achievements of those specialized maps.

**Comment by mboeringa** [comment]
Please note that implementing **area:highway** is only really useful if this includes proper rendering according to the **layer=x** tag. If this tag is not respected in the rendering, many complex highway junctions with multiple flyovers will become illegible random stackings of highway areas.

Additionally, it is likely **man_made=bridge** must be rendered according to layer=x as well. 

Both of these can be a major challenge to implement. I have succeeded in adding this functionality to my ArcGIS Renderer, and this was quite a job, but I don't know what the consequences are for the carto style...

**Comment by Marekkleciak** [comment]
This is good comment, that's true.
I would suggest colors:

Layer = -5 -> colour=#282828
Layer = -4 -> colour=#353535
Layer = -3 -> colour=#434343
Layer = -2 -> colour=#505050
Layer = -1 -> colour=#5d5d5d
Layer = 1 -> colour=#6b6b6b
Layer = 2 -> colour=#787878
Layer = 3 -> colour=#868686
Layer = 4 -> colour=#939393
Layer = 5 -> colour=#a1a1a1
Layer = 6 -> colour=#aeaeae

This should be enough for most complicated situations.

**Comment by mboeringa** [comment]
> This is good comment, that's true.
> I would suggest colors:

I am not entirely sure what adding colors is going to solve. The features will still be stacked incorrectly if not processed and ordered by the style and/or osm2pgsql.

In addition, layer = 6 is not valid according to Wiki. I don't think expanding the allowed values is wise from a rendering point of view. Before you know it, people will tread **layer** as **level**, and demand layer = 99 to be rendered as well... The main point though is that each additional supported layer value also results in extra drawing cycles and thematic layers needed to be processed in a style and renderer, whether Mapnik or something like QGIS or ArcGIS. At some point, and with 11 valid layer values we may already have reached this point, this becomes wholly impractical to deal with.

E.g., for my ArcGIS Renderer, I renderer these 14 thematic layers in a layered fashion using OpenStreetMap's **layer = x** tag:
- Highway (Lines)
- Highway (Lines) - Bridge
- Highway (Lines) - Tunnel
- Highway (Lines) - Crossing
- Highway (Lines) - Large Scale
- Highway (Lines) - Large Scale - Bridge
- Highway (Lines) - Large Scale - Tunnel
- Railway (Lines)
- Area:highway (Polygons)
- Area:highway (Polygons) - Road Markings (e.g. buslane, emergency etc.)
- Highway (Lines) - Oneway Arrows
- Highway (Lines) - Oneway Arrows Dark (dark version of arrows)
- Highway (Lines) - Oneway Arrows Cycleway (arrows for cycleways only)
- Man Made (Polygons) (bridges, viaducts as areas)

With potentially 11 values for each thematic layer, this potentially results in 11x14 = 154 ArcGIS Feature Layers in the Table of Contents of ArcMap. That is already pretty gruesome... let alone with more values.

From a practical point of view, I am pretty sure the 11 values from -5 to 5 should cover all conceivable real world situations, as long as people stick to what **layer** was supposed to solve in the first place - a rendering issue for **outdoor** highways and railways - and not tread it as **level**.

**Comment by pnorman** [comment]
> I support adding it because it should help reduce misuse of `highway=*` + `area=yes`

Over the last years my support for adding this has become more reluctant. I would oppose adding it because I think it results in bad cartography and the view that a map is a vectorized aerial image, except it has the potential to reduce misuse of area=yes highway=* tagging

> With potentially 11 values for each thematic layer, this potentially results in 11x14 = 154 ArcGIS Feature Layers in the Table of Contents of ArcMap. That is already pretty gruesome... let alone with more values.

Some time ago we stepped away from the need to have code for each possible value of layer. This is a good example of why we're not stepping back. Anything that having code for each value poses an unacceptable maintenance burden on what are already some of the most complex layers in the style.

> Please note that implementing **area:highway** is only really useful if this includes proper rendering according to the **layer=x** tag

Except for some bugs and edge cases I think we've got the use of layer=* we want. What to draw on top of other objects is a cartographic choice, and we've made ours.

**Comment by Marekkleciak** [comment]
I would oppose adding it because I think it results in bad cartography and the view that a map is a vectorized aerial image, except it has the potential to reduce misuse of area=yes highway=* tagging
-&gt; Could you explain why it results in bad cartography? 1. Everywhere area:highway was drawn the result is better positioning of  midline vestor of the street, also the footways are collected more carefully.
2. The real problem is, that area:highway mapping is possible only in areas with very good resolution of aerial images. But we get such images step by step. We had the same problem with buildings some years before. The resolution of Bing was not good enough and the results were pretty bad. Now we can map more precisely. 
3. Area highway mapping made mostly very experienced mapper. And they are mostly carefully.
 With potentially 11 values for each thematic layer, this potentially results in 11x14 = 154 ArcGIS Feature Layers in the Table of Contents of ArcMap. That is already pretty gruesome... let alone with more values.
-&gt; Please look at this: http://www.darkroastedblend.com/2006/11/incomprehensible-intersections.html 
I tried to analyse how many levels are neccesarily. Maximum I saw is seven.
I made one mockup: http://wiki.openstreetmap.org/wiki/File:StudyAreaHighwayLevelsRendering.jpg It is just seven. 
I got it only by adding of black to the basic colour. So, if needed I can prepare color table suggestion for all elements.Some time ago we stepped away from the need to have code for each possible value of layer. This is a good example of why we're not stepping back. Anything that having code for each value poses an unacceptable maintenance burden on what are already some of the most complex layers in the style.Please note that implementing area:highway is only really useful if this includes proper rendering according to the layer=x tagExcept for some bugs and edge cases I think we've got the use of layer=* we want. What to draw on top of other objects is a cartographic choice, and we've made ours.—

**Comment by mboeringa** [comment]
> Some time ago we stepped away from the need to have code for each possible value of layer. This is a good example of why we're not stepping back. **Anything that having code for each value poses an unacceptable maintenance burden on what are already some of the most complex layers in the style.**

Totally agree, and this why I developed an automated routine to generate each ArcGIS Feature Layer for a particular OSM **layer = x** value from a single base ArcGIS layer file as a copy of that base layer. So I essentially automatically "clone" the symbology from a base layer file. This means users only have to maintain symbology once for each thematic layer, so in the example I gave, there are just 14 base ArcGIS layer files to maintain, not 154 for each permutation of thematic feature layer and OSM layer value...

Since I auto-generate the copies of the ArcGIS feature layer (which only exist as part of an ArcMap document, not on disk), _from a purely technical point of view_, it would be no problem to support e.g. 100 OSM layer values (0..99).

I'll leave it to your imagination what the practical consequences would be of trying to render 1400+ layers...

> I tried to analyse how many levels are neccesarily. Maximum I saw is seven.

So, basically, with these extreme examples, we are still safe with 11 available OSM layer values (-5 to 5).

In fact, from a practical point of view, it is a misconception that to render the data properly visually, you need to make all flyovers their own OSM layer value. E.g., if you look at the underlying data from this Atlanta junction:

http://www.openstreetmap.org/#map=17/33.61748/-84.48647

you will notice that despite an apparent 3 or 4 road "levels", the highest OSM layer value used is just **layer = 2!**. This is a good example of proper usage of what layer was supposed to do: solve a render issue, not define absolute heights or levels, by only using a higher layer value for the extent two road lines cross, not over the entire length of a flyover (which isn't necessarily wrong, but may lead to technically unneeded higher maximum layer values, in the sense of not needed for proper visual rendering).

**Comment by Marekkleciak** [comment]
Exactly. In my opinion different colours for level=* are not neccesary if we used outline shapes for area:highway elements. See mockup:

http://wiki.openstreetmap.org/wiki/File:MockupAreaHighwayWithStreetOutlines.jpg

**Comment by dieterdreist** [comment]
2016-09-16 11:18 GMT+02:00 Marekkleciak <notifications@github.com>:

> Exactly. In my opinion different colours for level=* are not neccesary if
> we used outline shapes for area:highway elements. See mockup:




+1, (thin) outlines are desirable for these areas, at least in certain
(high) zoom levels, they can show the borders of them (e.g. curbs) also in
case there are many adjacent areas (and we decided to use the same color
for all of them)

**Comment by polarbearing** [comment]
area:highway is not even an agreed mapping method, it is used in two separate proposals which are immature and contain questionable tags and analogies, see the dispute about [area:highway=emergency](https://wiki.openstreetmap.org/wiki/Talk:Proposed_features/Street_area#area:highway.3Demergency) or [Plumbing principles](https://wiki.openstreetmap.org/wiki/Talk:Proposed_features/Street_area#The_.27plumbers_principle.27). That needs to be sorted first. 

From my perspective it is mostly a playground for couch mappers who have already drawn all building outlines in their area and want to vectorise more from their aerial imagery, with limited usability.

**Comment by Marekkleciak** [comment]
Well, I assume, the most of mappers do already their work using aerial imagery.  
Regarding a:h=emergency: I wrote in the link above my point of view.
In fact, we have in the zoom level 19 rendering of a lot of more or less useful details but exactly the streets, and we have the name OpenSTREETmap have no good representation.

Please look at this: http://www.openstreetmap.org/#map=19/48.87373/2.29567
The mapper collects already the most information needed for rendering of area:highway but we don´t do it. 

There is also an other point: Please keep in mind, by all differences, about "wikipedia effect". In areas, where everything is collected, mapper stops to map. Extending of map rendering with area:highway would give us all new impulses.

**Comment by d1g** [comment]
> area:highway is not even an agreed mapping method

@polarbearing AFAIK there no second proposal that actually **tries to replace or enhance** Marekkleciak efforts. 

Proposal from flaimo merely states "uh oh we don't have tagging for 2D roads" with area:highway=* as **the only tag** in scheme: https://wiki.openstreetmap.org/wiki/Proposed_features/area:highway#Values - it is unclear what should we do about priority (major/minor service road) or with pedestrian areas.

To me, [it is the only method](https://wiki.openstreetmap.org/wiki/Proposed_features/Street_area). We need at least one tagging scheme for highways as areas. Details how we should tag splits roads/footways on crossings can be improved over time.

> or Plumbing principles. That needs to be sorted first. 

"Questionable analogies" by Imagico has nothing to do with tagging/carto style IMO. But yes, feel free to improve descriptions/language/wiki: "Road crossings are separate from roads" - is it better than "[Plumbers principle](https://wiki.openstreetmap.org/wiki/Proposed_features/Street_area#Plumbers_principle)"?

**Comment by mboeringa** [comment]
> it is unclear what should we do about priority (major/minor service road) or with pedestrian areas.

In my personal opinion, there is very little use of rendering road classification (as in motorway,trunk,primary etc.) at the maximum zoom levels where **area:highway** rendering is applicable. Road classification is most useful at mid to small scales.

Instead, in my ArcGIS Renderer, I actually render area:highway indirectly: in reality, I render **"surface=x"**, **_but only for those features additionally tagged with area:highway_**.

Showing **surface=x** is exactly what most large scale (cadastral) maps do. They don't usually show the road classification at all, think of detailed CAD drawings of a planned new neighbourhood, but surface types (set, concrete, asphalt etc.) instead.

By rendering **surface=x** combined with **area:highway=x**, there is a nice positive incentive to tag **_both_**, which really should enhance the usefulness of this data in the OSM database for different purposes.

**Comment by polarbearing** [comment]
I'm not against the area:highway in general, it just has to be consolidated into an agreed mapping style first, avoiding bizarre extremes as tagging the angle of the hatch painting of the road. 

My fear is that it might tempt some people to start mapping every segment of the white lines.

And yes, taking analogies e.g. from graph theory is better than from plumbing.

**Comment by d1g** [comment]
> as tagging the angle of the hatch painting of the road

It depends whatever local highway code has rule for it. If yes, we potentially can use this data to simulate traffic or at least to have less uncovered questions "how to tag X from highway code". Real problem with that is that highway code (read: infrastructure) is different across countries.

**Comment by Marekkleciak** [comment]
Paul, look at this: http://taginfo.openstreetmap.org/tags/area=yes#combinations
highway=* with area=yes ->240 314 elements
highway=footway with area=yes34 766 elements

You wrote: we don´t strive to be a reproduction of aerial photo. May be, but many users do exactly this using wrong tagging which disturb the most of routing engines.

**Comment by d1g** [comment]
> We don't strive to be a reproduction of an aerial photo

You simply can't reproduce real world without "colour" information (or very low-level details tagged by  [material science](https://en.wikipedia.org/wiki/Materials_science) experts: too many reasons about why paving stones may/have different colours.

All fresh asphalt looks the same, give it 10 years in rural area and it will be hard to distinguish from the surrounding soil.

If you render everything as fresh asphalt in dark grey/black then you go unrealistic route already IMO: making all roads eye-candy when they (or some of them) are unmaintained IRL.

**Comment by Marekkleciak** [comment]
I agree with that as well.

**Comment by dieterdreist** [comment]
2016-09-19 2:15 GMT+02:00 Paul Norman <notifications@github.com>:

> A map chooses what data to represent and is a representation of the world.
> We don't strive to be a reproduction of an aerial photo, which is mostly
> what I've seen with area:highway examples.




it is your interpretation that people mapping precise geometry of road
surfaces are striving to reproduce aerial imagery. As aerial imagery is
reproducing (more or less) faithfully the actual geometry of the real
world, it is clear that some resemblance between them and a mapped road
surface occur (they are both representing the same thing). It is a question
how you want to use the data which representation works better for you, and
for some uses you will have great benefit from a faithful surface
representation, e.g. when looking at sealed surfaces, when doing surface
routing, etc. while for others you would need a graph model. For nice high
zoom _rendering_ it is surely beneficial to have actual surface shapes,
hence this style should (IMHO) support it.

**Comment by kocio-pl** [comment]
> If you render everything as fresh asphalt in dark grey/black then you go unrealistic route already IMO: making all roads eye-candy when they (or some of them) are unmaintained IRL.

It will simply be closer to reality than currently. Water in the river may not always be blue, but so what?

I see it from a different perspective: how much more realistic is red/yellow/white line with constant width at z19?

**Comment by d1g** [comment]
> I see it from a different perspective: how much more realistic is red/yellow/white line with constant width at z19?

Many users said that highway classification by importance (this is what rendered using different colours in highways=*) is meant to quickly orient yourself in unknown city. It is not good for routing defaults (compared to surface), low level rendering e.t.c

Well we can disable ordinary linear roads at z19, they are not able to capture uneven widths of the roads (parking, lanes, other reasons)

**Comment by mboeringa** [comment]
> **Well we can disable ordinary linear roads at z19**, they are not able to capture uneven widths of the roads (parking, lanes, other reasons)

I don't think that is an option. It would mean missing roads for large parts of the globe at Z19. Maybe a switch at a future Z20 zoom level (currently not available), would be an option, but I solved it by overlaying the **area:highway** features on top of the line rendering in my personal renderer. 

That, however, requires the line-version road widths to be less than the average comparable **area:highway** feature of the same highway class to completely hide the line representation. That is probably not the case in Carto, as Carto uses pretty wide roads at high zoom. In my personal renderer, linear ways are admittedly much narrower.

I actually **_need_** the linear representation below the area:highway features to still provide the street name labelling, although I could potentially solve this by creating a new layer without symbolization but with labels, but this way I keep having (linear) roads at highest zoom levels even where **area:highway** is missing (which is by far the majority of the globe of course), they just disappear below the corresponding **area:highway** features.

As you can see, even from a simple cartographic point of view, it is not such an easily solved issue and also heavily depends on the base style already in use upon implementation.

Anyway, these are samples from my ArcGIS Renderer to give some ideas. Notice as well that in the third image, **area:highway = x** and **man_made = bridge** features honour the OMS **layer = x** tag.:

![arcgis_renderer_for_openstreetmap-area_highway3](https://cloud.githubusercontent.com/assets/7635726/18645473/a0df1664-7ead-11e6-9eec-22e93556a4ef.jpg)
![arcgis_renderer_for_openstreetmap-area_highway4](https://cloud.githubusercontent.com/assets/7635726/18645474/a0e4ff7a-7ead-11e6-96d0-31ee637ca826.jpg)
![arcgis_renderer_for_openstreetmap-area_highway5](https://cloud.githubusercontent.com/assets/7635726/18645475/a0e66040-7ead-11e6-908b-791d38b5d734.jpg)
![arcgis_renderer_for_openstreetmap-area_highway6](https://cloud.githubusercontent.com/assets/7635726/18645476/a0e7a7b6-7ead-11e6-88a0-09ccba73042d.jpg)

**Comment by Marekkleciak** [comment]
congratulations! It looks really good...
Dnia 19 września 2016 21:17 mboeringa napisał(a):
 Well we can disable ordinary linear roads at z19, they are not able to capture uneven widths of the roads (parking, lanes, other reasons)I don't think that is an option. It would mean missing roads for large parts of the globe at Z19. Maybe a switch at a future Z20 zoom level (currently not available), would be an option, but I solved it by overlaying the area:highway features on top of the line rendering in my personal renderer.That, however, requires the line-version road widths to be less than the average comparable area:highway feature of the same highway class to completely hide the line representation. That is probably not the case in Carto, as Carto uses pretty wide roads at high zoom. In my personal renderer, linear ways are admittedly much narrower.I actually need the linear representation below the area:highway features to still provide the street name labelling, although I could potentially solve this by creating a new layer without symbolization but with labels, but this way I keep having (linear) roads at highest zoom levels even where area:highway is missing (which is by far the majority of the globe of course), they just disappear below the corresponding area:highway features.As you can see, even from a simple cartographic point of view, it is not such an easily solved issue and also heavily depends on the base style already in use upon implementation.Anyway, these are samples from my ArcGIS Renderer to give some ideas. Notice as well that in the third image, area:highway = x and man_made = bridge features honour the OMS layer = x tag.:

**Comment by Marekkleciak** [comment]
User Tomasz_W sent me this:
http://wiki.openstreetmap.org/wiki/File:Tomasz_W_Bridge_area_mockup1.png
http://wiki.openstreetmap.org/wiki/File:Tomasz_W_Bridge_area_mockup2.png
http://wiki.openstreetmap.org/wiki/File:Tomasz_W_Tunnel_area_mockup1.png
http://wiki.openstreetmap.org/wiki/File:Tomasz_W_Tunnel_area_mockup2.png
http://wiki.openstreetmap.org/wiki/File:Tomasz_W_Tunnel_area_mockup3.png

He is thinking abour better visualisation of bridges and tunnels from area:highway point of view.

**Comment by pnorman** [comment]
We talked about this at the SOTM hack day and decided to close the issue

- Implementing this PR by rendering all objects of all types by layer tag is out of scope of this issue, and not an option for technical reasons. It is **not** practical, even if it is desirable.

- If we were to implement this PR, it would be a simple rendering, probably just rendering the area in the colour corresponding to fills. None of the examples fit well with the rest of the OpenStreetMap Carto style.

- Making a map is about what data not to include as much as what to include.

- Rendering highways mapped as areas obscures the bridge and other topological information. Transparency is not an option.

**Comment by mboeringa** [comment]
I appreciate the difficulties of implementing something like this, probably as few others, having developed and implemented area:highway renderer in my own ArcGIS renderer. It was only after some careful deliberation, that I realized my models could probably be adapted to do this.

Nonetheless, I am wonder what PR you are referencing? I think this issue was just about exploring the potential for it. I don't see any concrete PR, just ideas and some (mock-up) examples, including my own.

**Comment by Marekkleciak** [comment]
1. "Implementing this change by rendering all objects of all types by layer tag is out of scope of this issue, and not an option for technical reasons. It is not practical, even if it is desirable."

-> Exactly. And it is not necessarily.  We need only one color value with outline for the following elements: street as general, footway, cycleway, bus stop, taxi stop, "emergency" or "dashed" areas. 

2. "If we were to implement this change, it would be a simple rendering, probably just rendering the area in the colour corresponding to fills. None of the examples fit well with the rest of the OpenStreetMap Carto style." 

-> Yes, we should implement as the first step a simple rendering of area in the colour corresponing to fills.  Examples: 

https://wiki.openstreetmap.org/wiki/File:MockupAreaHighwayWithStreetMidlines.jpg 
https://wiki.openstreetmap.org/wiki/File:MockupAreaHighwayWithStreetMidlinesBright.jpg

3. "Making a map is about what data not to include as much as what to include." Yes, we speak only about the highest zoom level. There is an important aspect: We don´t show turn:lanes as well. The polish rendering, let´s say here: http://osmapa.pl/w/area/?lat=51.76106&lon=19.48626&zoom=18&ol=Qq show that. You should go in the left corner down and use all options except the last both.

4. "Rendering highways mapped as areas obscures the bridge and other topological information. Transparency is not an option."
-> This is the problem we have always, when we render a map. See e.g. here:
http://www.openstreetmap.org/#map=19/22.28453/114.19181 
I suggest use of area:highway rendering for the highest zoom level only.

**Comment by d1g** [comment]
@pnorman,

**Comment by d1g** [comment]
@pnorman, Paul, even **without colours** we would be happy to see area:highways=* rendered (even most popular subset of it's classes)

> If we were to implement this change, it would be a simple rendering, probably just rendering the area in the colour corresponding to fills.

I would happy to see **just outlines** of such areas (without fills, without coloured fills)
Similar to kerbs/berriers and other infrastructure https://github.com/gravitystorm/openstreetmap-carto/issues/180#issuecomment-247914821

**Comment by d1g** [comment]
@pnorman, I would say more if area:higway=* will get support in openstreetmap-carto I would expect drop in [area=yes together together with highway=*](http://taginfo.openstreetmap.org/tags/area=yes#combinations).

area:higway was never supported by openstreetmap-carto, right?

**Comment by ZLima12** [comment]
This seems to have been forgotten about. I don't think that the average user would consider this to be too much detail, especially if it is only rendered at the highest zoom level. In some public parks, for example, this could add a nice level of detail.

**Comment by imagico** [comment]
One option to address this (kind of) would be to render area:highway polygons in a uniform styling in (or immediately above) the landcover layer.  That would be a bit like #3872 (which we gave up on highway polygons) but with uniform styling and only for the area:highway polygons.

This way we would be able to provide visual feedback on the mapping but without running into the various problems mentioned above.

**Comment by jeisenbe** [comment]
> without running into the various problems mentioned above.

Would there be no casing? How would we handle situations where multiple layers of `area:highway=` are mapped on top of each other, e.g. where there are bridges or tunnels?

**Comment by dieterdreist** [comment]
sent from a phone

> On 4 Feb 2021, at 02:43, Joseph E <notifications@github.com> wrote:
> Would there be no casing? How would we handle situations where multiple layers of area:highway= are mapped on top of each other, e.g. where there are bridges or tunnels?


in architecture, when there are hidden lines which you want to show (and this is important, because all hidden lines would often be too much), you draw them above, but in a dashed style. This style draws tunnels below (necessary because they also have a fill) which leads to situations where they are completely covered.
Christoph’s suggestion would work, but would lead to multiple lines for the same road, which would lead potentially to confusion or ugly rendering in some zoomlevels (appearing as “double casing”)(resulting from (even partial) overlap of highway lines and areas.
Rendering areas between highway fills and labels (including symbols like oneway and stop), and only in highest zoom levels, would IMHO lead to a better looking result, particularly when sidewalks and carriageway are represented with individual areas, or explicit kerbs are drawn. The higher level of detail would overrule the highway centre line display (potential problem could be people refraining from fixing the highway grid because it visually works with areas alone)

**Comment by dieterdreist** [comment]
sent from a phone

> On 4 Feb 2021, at 02:43, Joseph E <notifications@github.com> wrote:
> 
> How would we handle situations where multiple layers of area:highway= are mapped on top of each other, e.g. where there are bridges or tunnels?

the areas should respect explicit layer tags, as should bridges and buildings ;-)

**Comment by imagico** [comment]
> Would there be no casing? How would we handle situations where multiple layers of area:highway= are mapped on top of each other, e.g. where there are bridges or tunnels?

It would all be flatly rendered in uniform color in or directly above the landcover layer so there would be no layering issues, artefacts due to interference with linear highway mapping or added complexity in the already complex roads layers.

An overall outset casing would be possible but IMO probably not desirable - it would hardly provide additional feedback, would add noise and could be prone to confusion and ugly iterference with line feature rendering like barriers.

The idea would be to acknowledge and provide basic feedback on the data that is being entered but equally acknowledge and not try to ignore that neither consistency in established mapping conventions not our technical constraints here in combination with our goals allow more specific rendering.

**Comment by dieterdreist** [comment]
sent from a phone

> On 4 Feb 2021, at 12:57, Christoph Hormann <notifications@github.com> wrote:
> 
> It would all be flatly rendered in uniform color in or directly above the landcover layer so there would be no layering issues, artefacts due to interference with linear highway mapping


in lower zoom levels these would usually be covered by the linear highways, but in levels like 18 I imagine artefacts due to linear highways covering partly or almost the road areas, you would have tiny bits visible here and there below the highways and their casing.

**Comment by Marekkleciak** [comment]
We could use color table with different grey colors for area:highway=<m> and layer =< n>
For instance: 
![image](https://user-images.githubusercontent.com/1079654/106897130-cea57200-66f2-11eb-8747-164f13a9ded9.png)
If somebody her could try the look in one complicated situation?

**Comment by imagico** [comment]
> in lower zoom levels these would usually be covered by the linear highways, but in levels like 18 I imagine artefacts due to linear highways covering partly or almost the road areas, you would have tiny bits visible here and there below the highways and their casing.

That depends on the highway type and latitude.  For footways/tracks the polygon rendering would be visible fairly early already - like in case of the example mentioned in #4316.  In case of larger roads it would be visible starting at z16-z18 depending on latitude.  It would not be more of an artefact than with existing landcover rendering where it is not mapped up to the road center so it shows as negative space.

**Comment by matkoniecz** [comment]
> We could use color table with different grey colors for area:highway= and layer =< n>

This is problematic as layer tag is solely about relative layer of object at the same location. This would unilaterally redefine this tag to have different semantics.

**Comment by ZLima12** [comment]
If anyone needs a good test case for this, see [this park that I mapped](https://www.openstreetmap.org/#map=17/41.20833/-73.89902). I think it would look great here.

**Comment by krolinventions** [comment]
@pnorman Sorry for creating a new issue, I though the others which are years/decades old would perhaps be outdated.

What would the next steps for me be if I would want to pursue this (and put some effort in)?

Would you accept a PR? I would not mind giving it a try even if it turns out that it would have to be declined (for example, because it turns into a mess).

Would I have to find consensus on the way this is mapped or are we fine in this regard?

Is there another place I should go to discuss this?

**Comment by imagico** [comment]
> What would the next steps for me be if I would want to pursue this (and put some effort in)?

To bring forward any new arguments on the matter here without reiterating discussions we already had.  In addition to the discussion above there is significant further context to consider, a summary of some of it can be found in https://github.com/gravitystorm/openstreetmap-carto/issues/172#issuecomment-1332262761.  Other aspects are mentioned in https://github.com/gravitystorm/openstreetmap-carto/issues/4948#issuecomment-2034152150

Generally speaking the OSM-Carto road rendering system is complicated - but despite this complexity has numerous limitations.  A discussion of most of this can be found in https://imagico.de/blog/en/navigating-the-maze-part-1/.

> Would you accept a PR?

I would strongly suggest to discuss whatever concept you have for rendering before investing into a concrete implementation.  Various arguments regarding various ideas have already been discussed so make sure you take them into consideration.

> Would I have to find consensus on the way this is mapped or are we fine in this regard?

> Is there another place I should go to discuss this?

A renewed look what the de facto semantics and use patterns of these tags would definitely be helpful, not only for OSM-Carto.  Documentation on the OSM wiki is largely consisting of prescriptive ideas on tagging rather than documentation on how tags are actually used.  Last time i looked a lot of  `area:highway` essentially implemented manual buffering of constant width roads while there does not seem any consensus (or attempts to build consensus) on if that is a suitable way of mapping or not.  In terms of aggregated use numbers all `area:highway` tags together are currently used 268k times.  Compared to that the `width` tag on `highway=*` (which we don't render either) is used 2.75M times.

By far the most commonly tagged value of `area:highway` is `footway` - which bears the question of the semantic delineation between `area:highway=footway` and `highway=footway`  on polygons (which we render - with the constraints of our current handling of road polygons).  This is especially the case since many of the `area:highway=footway` polygons in the database contain junctions.

Discussions on how things should be tagged and mapped should go to different places, ideally where a broad range of mappers can be reached.

**Comment by krolinventions** [comment]
@imagico Thank you! I've read those links and I'm not discouraged yet.

I would try the simplest possible solution. The area should be drawn just before (below) the road. I would not mind the way of the road showing over top of the road area. For cycle paths and footpaths this is completely fine and this will even make the map easier to read. A future enhancement could be to detect if there is road area and use a less obtrusive way of drawing the way itself.

I've tested highway=footway area=yes. Those don't show up for me. The only thing I can get to show up is highway=pedestrian area=yes. See https://www.openstreetmap.org/changeset/150052388#map=19/52.16667/4.47920 for an experiment with area highways.

I've done some initial research in some cities on the area=yes and the area:highway using overpass, and the vast majority of the area=yes usage is pedestrian zones. Which isn't surprising as there is consensus over those. This means that for all other highway types the area:highway is much more common.

It looks like that for half of Manhattan all the sidewalks are mapped as highway=pedestrian area=yes so maybe I should just use that for now and see if I get complaints. Changing the tags on an area isn't that much work anyway if the consensus shifts.

I'll report back if this works well enough for me or if I really need cycle paths and traffic islands to show up. Leaving the road itself as a void works fine for me, I don't care too much about cars anyway.

**Comment by dieterdreist** [comment]
sent from a phone

> On 15 Apr 2024, at 21:28, Gerard Krol ***@***.***> wrote:
> 
> I've done some initial research in some cities on the area=yes and the area:highway using overpass, and the vast majority of the area=yes usage is pedestrian zones. Which isn't surprising as there is consensus over those. This means that for all other highway types the area:highway is much more common.


it is not the same, area=yes means non linear traffic space, area:highway is not telling you this, it can also represent linear space.

**Comment by imagico** [comment]
> I would try the simplest possible solution. The area should be drawn just before (below) the road. 

Rendering all `area:highway=*` polygons flatly below all roads (essentially like a `landuse=road` mapping) has already been discussed - see https://github.com/gravitystorm/openstreetmap-carto/issues/180#issuecomment-772921484.

> I've tested highway=footway area=yes. Those don't show up for me. 

It quite clearly does:

https://imagico.de/map/styleinfo/#style=osmcarto&section=tags&key=highway&value=footway&ftype=polygon
https://www.openstreetmap.org/way/436187783

**Comment by krolinventions** [comment]
@imagico I see you mentioned it, but I don't see what problems with it were discovered? The replies to it look positive to me.

I wonder what went wrong with highway=footway area=yes for me. I probably just tagged it wrong.

@dieterdreist I know it's not the same - but it's not too wrong and at least it shows up. GraphHopper seems to route along the edges sometimes but that's not too distracting. Actually on a sidewalk you can walk wherever you want anyway. If Carto gets support for area:highway I'd definitely switch to that.

**Comment by imagico** [comment]
> I see you mentioned it, but I don't see what problems with it were discovered?

No one has so far made a compelling case how this would be beneficial for the goals of this project.  It was brought up as a potential way to avoid the various technical problems mentioned while still formally adding rendering support for the tags.  That does not mean it is a good idea.

The most important question you would need to answer is probably: What existing consistent mapping practice is the rendering supposed to support and is it realistic for the suggested design to actually do that?

**Comment by imagico** [comment]
So far i am not really seeing much in terms of arguments and evidence for a change in assessment on this issue.  Some of the main concerns in previous discussion are not even mentioned.

I last looked at the mapping practice in 2024.  Has mapping practice changed substantially since then and, if yes, how?  I can see that `area:highway=traffic_island` has now surpassed `area:highway=footway` as the most widely used value - together these two account for nearly half of the uses of the `area:highway` key.

Is `area:highway=traffic_island` a major concern?  If yes there would be not much point in rendering it below the road layers because it will hardly ever be visible in a meaningful way and it would be more confusing than helpful when it is (assuming the tag is used as documented - i have not checked).

Regarding `area:highway=footway` i wrote in 2024:

> By far the most commonly tagged value of area:highway is footway - which bears the question of the semantic delineation between area:highway=footway and highway=footway on polygons (which we render - with the constraints of our current handling of road polygons). This is especially the case since many of the area:highway=footway polygons in the database contain junctions.

Has this changed since 2024?  If yes - what is the de facto delineation?  For pedestrian you have:

* area:highway=pedestrian: 10k uses, including things like https://www.openstreetmap.org/relation/2571561, https://www.openstreetmap.org/relation/5558012, https://www.openstreetmap.org/way/557609536, https://www.openstreetmap.org/way/475634639, https://www.openstreetmap.org/relation/7455840, https://www.openstreetmap.org/way/117279296, https://www.openstreetmap.org/way/527550893
* linear highway=pedestrian: 500k
* highway=pedestrian polygons: 345k

This to me indicates that for pedestrian roads `area:highway=pedestrian` is practically nearly non-existant, let alone as a tag with distinct meaning.

The question of re-opening it for me depends on if it seems viable to render something (and that would be concrete tags, not generically a key) in a way that is beneficial to our goals both from the technical side and from the design side.  That is why my last previous comment here was:

> The most important question you would need to answer is probably: What existing consistent mapping practice is the rendering supposed to support and is it realistic for the suggested design to actually do that?

So far the best indication for a consistent mapping practice seems to be with `area:highway=traffic_island` - but i don't see a viable design concept for that so far.

**Comment by dieterdreist** [comment]
sent from a phone

> On 23 Dec 2025, at 19:35, Christoph Hormann ***@***.***> wrote:
> 
> that there are no similarly or more common methods to map the same real world concept


if there are many highways with width mapped, would it be an alternative in the sense of this requirement with regard to rectangular road areas? 
I have seen many instances of sidewalk areas mapped as highway=pedestrian area=yes, I don’t believe it is a good representation, but telling them to use area:highway=sidewalk doesn’t give them the feedback they want, and it’s hard to convince them. Similarly in Berlin, landuse is often extended beyond the parcels to the kerb line, presumably to make road area details appear in “standard maps”:  https://www.openstreetmap.org/#map=19/52.511276/13.405886

Generally, the sidewalk edges can be mapped as kerbs, the other edge is typically formed by landuse, buildings also provide spatial information (e.g. to represent the shape of narrow alleys, or the border as edges of building blocks and street areas). What remains between the kerbs is the highway area (render kerbs, encourage their mapping, they’re also interesting for other purposes). For detailed street representation, road markings are coming. https://taginfo.openstreetmap.org/keys/road_marking#chronology

**Comment by pnorman** [comment]
> ### The Case for Reconsideration
> Since the major decisions on this issue were made in 2016, the mapping landscape has evolved.

Your points don't address three of the reasons it was rejected in the past: complexity, cartography, and clutter.

**Comment by CanopusFalling** [comment]
@pnorman 

### Regarding Clutter 

osm-carto can already be tricked to essentially render `area:highway=*` by manipulating the surrounding landuse so the empty space covers the road. This is what it would look like if just the `area:highway=residential` (and maybe the trunk roads) were rendered.

<img width="1515" height="1018" alt="Image" src="https://github.com/user-attachments/assets/be70425b-4768-4bbf-8d6d-86ce28588022" />

Personally, I don't love mapping this way as it implies the pavement/sidewalk is part of the residential landuse and not its own feature, so I'd rather there was official support for it in Carto.

The output above doesn't look cluttered. If anything, it looks much better than my local "victim" neighborhood looks in Carto currently, where I haven't manipulated the landuse. The current rendering looks visually muddy and ironically more cluttered due to the thin areas between the road and the grass being rendered a different color.

<img width="1197" height="932" alt="Image" src="https://github.com/user-attachments/assets/c61ab8ad-cdc8-4592-8a71-2fb2796dd9e2" />

### Regarding Cartography

I much prefer the first output, which clearly shows the areas the roads occupy while not overwhelming the information. I actually find it works against the cartography of OSM Carto to ignore this information, leaving visual voids between buildings and the road in most places.

### Regarding Complexity

Could you clarify what you mean by complexity? I've attempted to address complexity issues and have read through this issue multiple times, but I can't find specific barriers that don't also apply to other things Carto already renders. Since multiple other renderers already support road areas, it seems like a solvable problem.

---

@imagico I'd be interested in your opinion on the two images above as well seeing as you asked for examples of what this might look like in practice.

I'd also like to point out that maintaining the status quo is still a choice with consequences. Mappers clearly feel the need to manipulate landuse to get a decent-looking output in Carto. While we tell people not to 'tag for the renderer,' if mapping the documented way results in a muddy or broken map, users will inevitably ignore the documentation to produce a visually useful result. The lack of support for `area:highway` inadvertently encourages this kind of data manipulation.

**Comment by ZLima12** [comment]
> Personally, I don't love mapping this way as it implies the pavement/sidewalk is part of the residential landuse and not its own feature, so I'd rather there was official support for it in Carto.

The sidewalks and streets arguably are part of the residential landuse, though this depends on the case and mapping style. For example, in a housing development, there should be one landuse=residential way that encloses the entire premises, sidewalks, streets, and all. That way would get the name=* for the development. In other cases, different mappers map this differently; some end the way at the street, others don't. Neither approach has been universally agreed upon.
